### PR TITLE
Use abi-gen in contracts and for events in 0x.js

### PR DESCRIPTION
**Pull request by LogvinovLeon**
This PR migrates to abi-gen contracts and 0x.js events

I still need to figure out how to do the wrapping in a less verbose way

## How Has This Been Tested?

All tests pass

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

* [ ] Change requires a change to the documentation.
* [ ] Added tests to cover my changes.

### CODE REVIEW

**Comment by LogvinovLeon** [comment]
Prettier doesn't suport TS 2.7 features yet and therefore fails to parse

File: packages/contracts/test/multi_sig_with_time_lock_except_remove_auth_addr.ts [hunk 1]
```diff
@@ -165,17 +168,17 @@ describe('MultiSigWalletWithTimeLockExceptRemoveAuthorizedAddress', () => {
             const txHash = await multiSigWrapper.submitTransactionAsync(validDestination, owners[0], dataParams);
```
Comment (fabioberger) [inline packages/contracts/test/multi_sig_with_time_lock_except_remove_auth_addr.ts]: Looks like `multiSigWrapper` is only used for calling  `submitTransactionAsync`. Is this not included in the auto-generated wrapper classes?

File: packages/contracts/util/exchange_wrapper.ts [hunk 2]
```diff
@@ -23,14 +25,14 @@ export class ExchangeWrapper {
...
             params.orderAddresses,
             params.orderValues,
             params.fillTakerTokenAmount,
             params.shouldThrowOnInsufficientBalanceOrAllowance,
-            params.v,
-            params.r,
-            params.s,
+            params.v as number,
+            params.r as string,
+            params.s as string,
```
Comment (fabioberger) [inline packages/contracts/util/exchange_wrapper.ts]: Why are these not automatically detected as being `number` and `string` types?

File: packages/testnet-faucets/src/ts/request_queue.ts [hunk 3]
```diff
@@ -46,9 +46,6 @@ export class RequestQueue {
             this._processNextRequestFireAndForgetAsync(recipientAddress);
         }, this._queueIntervalMs);
     }
-    protected _stop() {
-        clearInterval(this._queueIntervalId);
-    }
```
Comment (fabioberger) [inline packages/testnet-faucets/src/ts/request_queue.ts]: Why is this removed? This seems unrelated to this PR.

**Comment by fabioberger** [comment]
This looks great! 🥇 

What 2.7 features are you using @LogvinovLeon?

File: packages/contracts/src/contract_wrappers/generated/.gitignore [hunk 4]
```diff
@@ -0,0 +1,8 @@
+dummy_token.ts
+exchange.ts
+multi_sig_wallet_with_time_lock_except_remove_authorized_address.ts
+multi_sig_wallet_with_time_lock.ts
+multi_sig_wallet.ts
+token_registry.ts
+token_transfer_proxy.ts
+z_r_x_token.ts
```
[the remark this answers is not in the document]
Comment (LogvinovLeon) [reply packages/contracts/src/contract_wrappers/generated/.gitignore]: Yeah. I wasn't sure if I should care. This is formally correct conversion from ZRXToken to snake case. I'll special-case it.

File: packages/contracts/util/exchange_wrapper.ts (hunk 2 above)
Comment (LogvinovLeon) [reply packages/contracts/util/exchange_wrapper.ts]: In 0x.js we have SignedOrder and Order. In contracts we have Order with `v: number|undefined`. This is a topic for a future refactor. For now I just show the compiler that I'm sure that this order is signed at the moment.

File: packages/testnet-faucets/src/ts/request_queue.ts (hunk 3 above)
Comment (LogvinovLeon) [reply packages/testnet-faucets/src/ts/request_queue.ts]: This is never called in the codebase. With that removed - we have a strong guarantee that `this._queueIntervalId` is never undefined and therefore can promise that to a compiler

**Comment by LogvinovLeon** [comment]
@fabioberger https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#definite-assignment-assertions TS compiler checks that the interval has non-nullable type but it isn't assigned in the constructor, but it fails to see, that the constructor calls a function that assigns it so I make a hint.

File: packages/contracts/test/multi_sig_with_time_lock.ts [hunk 5]
```diff
@@ -52,20 +54,21 @@ describe('MultiSigWalletWithTimeLock', () => {
...
             before('deploy a walet', async () => {
-                multiSig = await deployer.deployAsync(ContractName.MultiSigWalletWithTimeLock, [
+                const multiSigInstance = await deployer.deployAsync(ContractName.MultiSigWalletWithTimeLock, [
                     owners,
                     SIGNATURES_REQUIRED,
                     0,
                 ]);
-                multiSigWrapper = new MultiSigWrapper(multiSig);
+                multiSig = new MultiSigWalletWithTimeLockContract(multiSigInstance);
+                multiSigWrapper = new MultiSigWrapper((multiSig as any) as MultiSigWalletContract);
```
[the remark this answers is not in the document]
Comment (LogvinovLeon) [reply packages/contracts/test/multi_sig_with_time_lock.ts]: MultiSigWrapper does coding of the parameters. We have no logic like that in auto-generated code

File: packages/contracts/test/multi_sig_with_time_lock_except_remove_auth_addr.ts (hunk 1 above)
Comment (LogvinovLeon) [reply packages/contracts/test/multi_sig_with_time_lock_except_remove_auth_addr.ts]: This function needs to code the params before passing them down

**Comment by LogvinovLeon** [comment]
Prettier fails because of that issue: https://github.com/prettier/prettier/issues/3561

File: packages/contracts/test/multi_sig_with_time_lock.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (fabioberger) [reply packages/contracts/test/multi_sig_with_time_lock.ts]: Can we move it elsewhere? The name of this class is awkward...

[the remark this answers is not in the document]
Comment (fabioberger) [reply packages/contracts/test/multi_sig_with_time_lock.ts]: ok, makes sense.

File: packages/contracts/util/exchange_wrapper.ts (hunk 2 above)
Comment (fabioberger) [reply packages/contracts/util/exchange_wrapper.ts]: ok. Please make sure to add a task for this refactor.

**Comment by fabioberger** [comment]
Ok, for now, let's not use `Definite Assignment Assertions` so we can push this out.
